### ISSUE TITLE
Should auto-correct of Lint/UnneededDisable remove the Enable comment as well?

### ISSUE THREAD

**Issue by ecbrodie** [rubocop-hq/rubocop#4528]
The goal of this issue is to explore whether the problem I am experiencing with Rubocop is actually a problem, and if the solution I am perceiving is desirable. If this issue is accepted, then I would like to personally attempt a PR to get this fixed. Thank you.

## Expected behavior

When `rubocop --auto-correct` detects a violation of the `Lint/UnneededDisable` rule, both the disable and the enable comments are removed, such that no trace of the unneeded disabled cop is left behind.

## Actual behavior

The disable comment is being removed, but not the enable comment. The means that a comment serving no purpose has been left behind.

## Steps to reproduce the problem

I ran into this issue recently during a refactoring of my Ruby code, which reduced the line length of one of my longer methods. This method had the `Metrics/MethodLength` cop disabled, since it was determined that a refactoring was not desirable at the time. After the refactoring was complete, then method looked something like this:

```ruby
# rubocop:disable Metrics/LineLength
def foo
  raise "this is for illustrative purposes only"
end
# rubocop:enable Metrics/LineLength
```

After running Rubocop with auto-correct, this source code becomes like this:

```ruby
def foo
  raise "this is for illustrative purposes only"
end
# rubocop:enable Metrics/LineLength
```

The enable line doesn't serve any purpose anymore, so this is treated as an unnecessary comment and removed manually. It would have been ideal if Rubocop removed that line automatically instead of me needing to manually check all files that Rubocop had changed.

My idea for a PR implementation is that when Rubocop removes a disable comment using the `Lint/UnneededDisable` cop, it looks for the very next enable comment of the same cop type and removes that line. Logic could be added in so that if the rule had not been enabled before the next disable, then there is no enable line to remove.

Like I said before, this whole issue is based on a problem that I perceive exists. I would love to submit a PR if I am right, but I am open to feedback and to understand more what the intent of the `Lint/UnneededDisable` cop is.

## RuboCop version

```
$ rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

**Comment by jonas054** [comment]
Late reply, but yes, that would be a good idea.

**Comment by bbatsov** [comment]
There hasn't been much activity on this ticket and our Core Team is spread too thin on so many tasks, so I'll just close it for the sake of having a cleaner lists of tasks to focus on. We'd gladly review a PR, but it's unlikely that'd we're going to tackle this ourselves in the foreseeable future.
